### PR TITLE
Add documentation for Flagger and Argo-Rollouts

### PR DESCRIPTION
**Pull request by silversoul93**
## Description

We analyzed, for a client, Flagger and argo-rollouts to use the blue/green deployment strategy. This documentation explains how to use them through the Mia-Platform console.

## Pull Request Type

- [X] Documentation content changes
- [ ] Bugfix / Missing Redirects
- [ ] Docusaurus site code changes

## PR Checklist

- [X] The commit message follows our guidelines included in the [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md#how-to-submit-a-pr)
- [X] All tests of Lint, cspell and check-content pass. ([How to launch them?](../blob/main/CONTRIBUTING.md#content-checks))
- [X] No sensible content has been committed

### CODE REVIEW

File: .cspell.json [hunk 1]
```diff
@@ -302,7 +302,9 @@
     "Noto",
     "Segoe",
     "kotlinx",
-    "reauthentication"
+    "reauthentication",
+    "argoproj",
```
Comment (fredmaggiowski) [inline .cspell.json:306]: Why this is required? Every occurrence seems to be between backticks; i'd try to fix the `ignoreRegExpList`, instead 🤷

File: .cspell.json [hunk 2]
```diff
@@ -302,7 +302,9 @@
     "Noto",
     "Segoe",
     "kotlinx",
-    "reauthentication"
+    "reauthentication",
```
Comment (fredmaggiowski) [inline .cspell.json:305]: this is not a word, the correct compound word should be re-authentication who added this 🤦

File: docs/paas/deployment_tools.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+:::info
+Blue/Green and Canary strategies can be mixed to allow some user to start using the new v2 version before the promotion (progressive traffic shifting).
+:::
+
+#### Tests automation
+
+Flagger executes a precise tests lifecycle, as shown in the previous paragraph, and provides different ways to automate tests.
+
+The main types are:
+* [**metrics analysis**](https://docs.flagger.app/usage/metrics): allows to specify one or more metrics to be analyzed to promote/rollback the release. Flagger supports different metrics providers (prometheus, dynatrace, datadog and so on);
```
Comment (fredmaggiowski) [inline docs/paas/deployment_tools.md:96]:
```suggestion
* [**metrics analysis**](https://docs.flagger.app/usage/metrics): allows to specify one or more metrics to be analyzed to promote/rollback the release. Flagger supports different metrics providers (Prometheus, Dynatrace, Datadog and so on);
```

These are all product names, and should be properly capitalized

File: sidebars.json [hunk 4]
```diff
@@ -1040,6 +1040,10 @@
             {
               "id": "paas/alarms",
               "type": "doc"
+            },
+            {
+              "id": "paas/deployment-tools",
```
Comment (fredmaggiowski) [inline sidebars.json:1045]: Why under paas? Is this something completely related to our paas infrastructure or cìan this documentation be available on any infrastructure?

File: docs/paas/img/argo-rollouts_dashboard_rollouts_list.png
Comment (fredmaggiowski) [inline docs/paas/img/argo-rollouts_dashboard_rollouts_list.png:1]: Where have this image been taken from? Is it a screenshot you made? are we sure we can post screenshot from other products? 🤔

File: docs/paas/img/flagger_hooks_types.png
Comment (fredmaggiowski) [inline docs/paas/img/flagger_hooks_types.png:1]: I'd not post a screenshot of another product documentation, let's enumerate them or (better) provide a link to the doc, without copying it on our side.

File: docs/paas/img/argo-rollouts_automatic_tests_yml.png
Comment (fredmaggiowski) [inline docs/paas/img/argo-rollouts_automatic_tests_yml.png:1]: Why the yaml in an image?

File: docs/paas/deployment_tools.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+id: deployment-tools
+title:  Deployment tools
+sidebar_label: Deployment tools
+---
+
+# Deployment tools
+
+Kubernetes offers, by design, two deployment strategies:
+
+* Recreate &rarr; terminates the pod and, once terminated, scales up the new one;
```
Comment (fredmaggiowski) [inline docs/paas/deployment_tools.md:11]: Why `&rarr;` generally we would simply use a colon `:`

File: docs/paas/deployment_tools.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+* Rolling Update (the default one) &rarr; the new pod is scaled up and, when ready and healthy, the old one is terminated;
+
+look at the [official documentation](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy) for more information.
+
+Sometimes the user needs other deployment strategies, not supported by design using Kubernetes.
+For this reason we analyzed some tool to fulfil this need:
+
+* [Flux - Flagger](https://flagger.app/);
```
Comment (fredmaggiowski) [inline docs/paas/deployment_tools.md:19]:
```suggestion
* [Flagger](https://flagger.app/) from Flux;
```

File: docs/paas/deployment_tools.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+* [Argo - Argo-rollouts](https://argo-rollouts.readthedocs.io/).
```
Comment (fredmaggiowski) [inline docs/paas/deployment_tools.md:20]:
```suggestion
* [Argo-rollouts](https://argo-rollouts.readthedocs.io/) from Argo.
```

File: docs/paas/deployment_tools.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+
+:::warning
+The tools above are not fully integrated into the Mia-Platform Console. The user can handle resources through the console and the projects configurations, but not through the Mia-Platform Console UI with a no-code approach.
```
Comment (fredmaggiowski) [inline docs/paas/deployment_tools.md:23]: What is the purpose of this warning?

File: docs/paas/deployment_tools.md [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+:::
+
+## Flux - Flagger
```
Comment (fredmaggiowski) [inline docs/paas/deployment_tools.md:26]:
```suggestion
## Flagger from Flux
```

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

File: sidebars.json (hunk 4 above)
Comment (silversoul93) [reply sidebars.json:1088]: It's not related to our PaaS Infrastructure. Where do you suggest to move it?

File: docs/paas/img/argo-rollouts_dashboard_rollouts_list.png
Comment (silversoul93) [reply docs/paas/img/argo-rollouts_dashboard_rollouts_list.png:1]: I did it. I really don't know, the Argo Rollout UI is part of the product, under Apache 2.0 license.

File: .cspell.json (hunk 1 above)
Comment (silversoul93) [reply .cspell.json]: I can confirm the `ignoreRegExpList` doesn't work. If I remove `reauthentication` the spellcheck fails even if the world is in triple backticks. I fixed it with a regex found here: https://github.com/streetsidesoftware/vscode-spell-checker/issues/202#issuecomment-377477473

File: .cspell.json (hunk 2 above)
Comment (silversoul93) [reply .cspell.json]: Removed.

File: docs/paas/img/argo-rollouts_dashboard_rollouts_list.png
Comment (silversoul93) [reply docs/paas/img/argo-rollouts_dashboard_rollouts_list.png:1]: @fredmaggiowski Who can I engage for this information?

File: docs/paas/deployment_tools.md (hunk 5 above)
Comment (silversoul93) [reply docs/paas/deployment_tools.md:11]: up

File: sidebars.json (hunk 4 above)
Comment (silversoul93) [reply sidebars.json:1088]: up

File: docs/paas/deployment_tools.md (hunk 8 above)
Comment (silversoul93) [reply docs/paas/deployment_tools.md:23]: The purpose is to warn the user that, at the moment, the Console UI doesn't have a dedicated page to graphically handle these tools configurations

File: docs/paas/deployment_tools.md (hunk 5 above)
Comment (fredmaggiowski) [reply docs/paas/deployment_tools.md:11]: it's not a matter of what I like lol, we just don't do it in any other bullet list in the repo :)

Comment (silversoul93) [reply docs/paas/deployment_tools.md]: I replaced the beautiful `&rarr;` with boring colons.

File: sidebars.json (hunk 4 above)
Comment (fredmaggiowski) [reply sidebars.json:1124]: let's keep it here; i'd think about a specific "third party integration" category in the infrastructure section

immagine


@mia-platform/sig-documentation  what do you think about this?

File: docs/paas/img/argo-rollouts_dashboard_rollouts_list.png
Comment (fredmaggiowski) [reply docs/paas/img/argo-rollouts_dashboard_rollouts_list.png:1]: Did you take the screenshot yourself or did you copy it from a third party?

**Review by fredmaggiowski — APPROVED** [review]

File: docs/paas/img/argo-rollouts_dashboard_rollouts_list.png
Comment (silversoul93) [reply docs/paas/img/argo-rollouts_dashboard_rollouts_list.png:1]: I took the screenshots by myself

Comment (fredmaggiowski) [reply docs/paas/img/argo-rollouts_dashboard_rollouts_list.png:1]: ok than it's fine

File: sidebars.json (hunk 4 above)
Comment (silversoul93) [reply sidebars.json:1140]: @fredmaggiowski I think you are right, a "third party integration" section could be useful for other tools too (e.g. kube-green). @mia-platform/sig-documentation Can I add this new section?

Comment (silversoul93) [reply sidebars.json:1140]: @mia-platform/sig-documentation up, I'd like to close this PR I worked on 😄

Comment (fredmaggiowski) [reply sidebars.json:1140]: @akelity

Comment (silversoul93) [reply sidebars.json]: @fredmaggiowski I met @akelity and we added the "Third Party Integration" section.

**Pull request merged**